backend/src/gh.rs

```diff
@@ -161,3 +171,242 @@ fn gen_jwt_token(client_id: &str) -> Result<String> {
+/// 
+/// # Parameters
+/// - `repo_url`: A string slice representing the full repository URL (e.g., 
+///   `https://github.com/owner/repository.git`).
+/// 
+/// # Returns
+/// A `Result<String>` where:
+/// - `Ok(<owner>/<repo>)`: A string in the format `<owner>/<repo>` representing the repository name and owner.
+/// - `Err(e)`: Returns an error if the `repo_url` is missing the expected format or `.git` suffix.
+/// 
+/// # Errors
+/// Returns an error if:
+/// - The URL does not contain both an owner and a repository name (e.g., `https://github.com`).
+/// - The URL does not match the expected pattern (missing or incorrect `.git` suffix).
+pub fn get_repo_name_from_url(repo_url: &str) -> Result<String> {
+    // Parse the repository name from the URL
+    let repo_path = repo_url
+        .trim_end_matches(".git")  // Remove the .git suffix
+        .rsplit('/')  // Split by '/'
+        .collect::<Vec<&str>>();  // Collect into a vector
+
+    // Ensure repo_path has both owner and repo
+    if repo_path.len() < 2 {
+        bail!("Invalid repo_url format, must be <owner>/<repo>.");
+    }
+
+    // Format as <owner>/<repo>
+    Ok(format!("{}/{}", repo_path[1], repo_path[0]))
+}
+
+/// Create a GitHub pull request using the provided parameters.
+///
+/// # Parameters:
+/// - `repo_url`: A string slice that contains the GitHub repository URL in the format
+///   `https://<host>/<owner>/<repo>.git`. This is used to derive the repository name.
+/// - `reqwest_client`: The `reqwest::Client` used to make HTTP requests.
+/// - `token`: The GitHub access token for authentication.
+/// - `head_branch`: The branch where changes are made.
+/// - `base_branch`: The base branch to which the pull request is opened.
+/// - `pr_title`: The title for the pull request.
+/// - `pr_description`: The description for the pull request.
+///
+/// # Returns
+/// A `Result<String>` containing the URL of the created pull request if successful, or an error
+/// if the pull request creation fails.
+///
+/// # Errors
+/// Returns an error if:
+/// - The `repo_url` is not in the expected format.
+/// - The pull request creation request fails (e.g., due to authentication issues or invalid input).
+pub async fn create_pull_request(
+    repo_url: &str,
+    reqwest_client: &reqwest::Client,
+    token: &str,
+    head_branch: &str,
+    base_branch: &str,
+    pr_title: &str,
+    pr_description: &str,
+) -> Result<String> {
+    // Get the repository name using the updated function
+    let repo_name = get_repo_name_from_url(repo_url)?;
+
+    // Prepare the JSON body for the pull request
+    let pr_body = json!( {
+        "title": pr_title,
+        "head": head_branch,  // The branch with the changes
+        "base": base_branch,  // The branch to merge into
+        "body": pr_description,
+    });
+
+    info!("Creating pull request to {}/repos/{}/pulls", GITHUB_API_URL, repo_name);
```

Comment:
This should probably be a debug statement, the api endpoint isn't useful info to have for most cases